### PR TITLE
Wip/hughsie/remote refresh client

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

**Comment by hughsie** [comment]
As an aside, fwupd_client_install_release() is something I also want to use from fwupdagent...

File: meson.build [hunk 1]
```diff
@@ -146,6 +146,9 @@ add_project_arguments('-DFWUPD_DISABLE_DEPRECATED', language : 'c')
 add_project_arguments('-D_BSD_SOURCE', language : 'c')
 add_project_arguments('-D_XOPEN_SOURCE=700', language : 'c')
 
+# needed for memfd_create()
+add_project_arguments('-D_GNU_SOURCE', language : 'c')
```
Comment (superm1) [inline meson.build:150]: Rather than `_BSD_SOURCE` and `_GNU_SOURCE`, can you just set `_DEFAULT_SOURCE`?

**Comment by superm1** [comment]
>As an aside, fwupd_client_install_release() is something I also want to use from fwupdagent...

An agent with super powers!

File: meson.build (hunk 1 above)
Comment (hughsie) [reply meson.build:150]: No, it really needs to be `_GNU_SOURCE`

**Pull request merged**